### PR TITLE
Added 65SC02 support for simulator.

### PR DESCRIPTION
**Pull request by IrgendwerA8**

### CODE REVIEW

File: src/sim65/6502.c [hunk 1]
```diff
@@ -205,7 +213,22 @@ int PrintCycles;
     unsigned Addr;                                              \
     Cycles = 5;                                                 \
     ZPAddr = MemReadByte (Regs.PC+1);                           \
-    Addr = MemReadZPWord (ZPAddr) + Regs.YR;                    \
+    Addr = MemReadZPWord (ZPAddr);                              \
+    if (PAGE_CROSS (Addr, Regs.YR))                             \
```
Comment (oliverschmidt) [inline src/sim65/6502.c]: The style in this file is to have curly brackets even for single instruction if-bodies.

File: src/sim65/6502.c [hunk 2]
```diff
@@ -362,6 +389,21 @@ static void OPC_6502_01 (void)
...
 
 
+static void OPC_65SC02_04 (void)
+/* Opcode $04: TSB zp */
+{
+    unsigned char ZPAddr;
+    unsigned char Val;
+    Cycles = 5;
+    ZPAddr = MemReadByte (Regs.PC+1);
+    Val    = MemReadByte (ZPAddr);
```
Comment (oliverschmidt) [inline src/sim65/6502.c]: The style of this file is to not have alignments of consecutive assignments (and you don't have them at other newly added code).

File: src/sim65/6502.c [hunk 3]
```diff
@@ -205,7 +213,22 @@ int PrintCycles;
     unsigned Addr;                                              \
     Cycles = 5;                                                 \
     ZPAddr = MemReadByte (Regs.PC+1);                           \
-    Addr = MemReadZPWord (ZPAddr) + Regs.YR;                    \
+    Addr = MemReadZPWord (ZPAddr);                              \
+    if (PAGE_CROSS (Addr, Regs.YR))                             \
+        ++Cycles;                                               \
```
Comment (oliverschmidt) [inline src/sim65/6502.c]: The style in this file is to have curly brackets even for single instruction if-bodies.

File: src/sim65/6502.c [hunk 4]
```diff
@@ -234,6 +257,8 @@ int PrintCycles;
             }                                                   \
             TEST_CF (Regs.AC);                                  \
             SET_OF ((res < -128) || (res > 127));               \
+            if (CPU!=CPU_6502)                                  \
```
Comment (oliverschmidt) [inline src/sim65/6502.c]: According to the style in this file this would be `CPU != CPU_6502` (with spaces).

**Comment by IrgendwerA8** [comment]
Done. I've still my problems with GitHub. Where were the style information and the complaints available? Navigating from a current commit to the follwoing Travis build seems to me good hidden too...

**Comment by oliverschmidt** [comment]
> Where were the style information and the complaints available? 

I must admit that I don't understand you well. I was referring to the sections on this very page (https://github.com/cc65/cc65/pull/405) as visible in the screenshots below:

a
b
c

**Comment by oliverschmidt** [comment]
> Navigating from a current commit to the follwoing Travis build seems to me good hidden too...

On this very page (https://github.com/cc65/cc65/pull/405) there's:

d

Alternatively you can follow the link e present on https://github.com/cc65/cc65 and http://cc65.github.io/cc65/#Links

Or do I misunderstand you?

File: src/sim65/6502.c [hunk 5]
```diff
@@ -215,7 +215,9 @@ int PrintCycles;
     ZPAddr = MemReadByte (Regs.PC+1);                           \
     Addr = MemReadZPWord (ZPAddr);                              \
     if (PAGE_CROSS (Addr, Regs.YR))                             \
+    {                                                           \
```
Comment (oliverschmidt) [inline src/sim65/6502.c]: Really sorry to bother you but the open curly bracket is supposed to be at the end of the previous line.

Comment (IrgendwerA8) [inline src/sim65/6502.c]: Fixed that too. But that is what I mean: Where can I find this "style guide" or is there an automatic check where I can see when I break it? (I'm familiar with "Resharper" in this respect)
(Not that I find the style very consistent nor I would prefer it... ;) )
You making others really a hard time to contribute when even copy & paste of existing code leads to refinishing...

Comment (oliverschmidt) [inline src/sim65/6502.c]: > But that is what I mean: Where can I find this "style guide"

It's not complete but it's at https://github.com/cc65/wiki/wiki/C-and-Assembly-sources-style-guide

> or is there an automatic check where I can inspect the violations when I break it? (I'm familiar with "Resharper" in this respect)

No, there isn't. 

> (Not that I find the style very consistent nor I would prefer it... ;) )

As far as I know the C sources are VERY consistent (in contrast to the Asm sources). I personally don't like the style but I didn't introduce it.

> You making others really a hard time to contribute when even copy & paste of existing code leads to refinishing...

I'm sorry for that. Can you please point out where existing C code doesn't adhere to the things I mentioned?

Comment (IrgendwerA8) [inline src/sim65/6502.c]: This
https://cloud.githubusercontent.com/assets/2664009/24075733/2453022a-0c21-11e7-99e4-a746bb66987d.png
...was reasoned by C&P, but as you may noticed I changed in the mean time all occurrences... (hopefully)

**Comment by oliverschmidt** [comment]
Thanks for your patience !

**Pull request merged**